relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -181,6 +182,48 @@ def bmo_file_loan_bug(self, loanid, slavetype, *args, **kwargs):
         self.retry(exc=exc)
 
 
+@task(bind=True)
+@add_to_history(
+    after="Waiting for a human to perform {kwargs[action_name]} (id {retval!s})")
+def register_action_needed(self, loanid, action_name):
+    if not action_name:
+        raise ValueError("must supply an action name")
+    try:
+        session = current_app.db.session('relengapi')
+        l = session.query(Loans).get(loanid)
+        if action_name == "add_to_vpn":
+            action_message = (
+                "Add user (%s) and machine (%s) to the VPN. "
+                "Following https://wiki.mozilla.org/ReleaseEngineering/How_To/Update_VPN_ACL"
+                % (l.human.ldap, l.machine.fqdn)
+            )
+        else:
+            raise ValueError("Invalid action name")
+        action = ManualActions(for_loan=l,
+                               timestamp_start=tz.utcnow(),
+                               msg=action_message)
+        session.add(action)
+        session.commit()
+        return action.id
+    except ValueError:
+        raise
+    except Exception as exc:
+        self.retry(exc=exc)
+
+
+@task(bind=True, max_retries=None, default_retry_delay=60)
+@add_to_history(
+    after="Noticed that a human performed pending action (id {args[1]}), continuing")
```

Comment:
Well actually, it is polling for results (polling db for result of the PUT endpoint).